src/wallet/wallet.cpp

```diff
@@ -4543,8 +3827,12 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
 
         // No need to read and scan block if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        if (walletInstance->nTimeFirstKey) {
-            if (Optional<int> first_block = locked_chain->findFirstBlockWithTimeAndHeight(walletInstance->nTimeFirstKey - TIMESTAMP_WINDOW, rescan_height, nullptr)) {
+        int64_t time_first_key = std::numeric_limits<int64_t>::max();
+        for (auto spk_man : walletInstance->GetActiveScriptPubKeyMans()) {
+            std::min(time_first_key, spk_man->GetTimeFirstKey());
```

Comment:
In commit "[ci skip] Use GetTimeFirstKey instead of nTimeFirstKey" (4b7effe37f5efeddb54523b95e58d464bc317e7f)

Is this a bug? It seems like this should say

```
time_first_key = std::min(time_first_key, spk_man->GetTimeFirstKey());
```

If this is a bug, I'm a little surprised there isn't any test coverage for this.

Reply:
Seems like a bug, fixed. Changing it has no effect on any tests...